recipes/kwidgetsaddons/all/conanfile.py

```diff
@@ -0,0 +1,76 @@
+import os
+from conans import ConanFile, CMake, tools
+
+
+class KwidgetsaddonsConan(ConanFile):
+    name = "kwidgetsaddons"
+    license = ("BSD-3-Clause", "LGPL-2.0-only", "LGPL-2.1-only", "LGPL-3.0-only", "LicenseRef-KDE-Accepted-LGPL", 
+        "GPL-2.0-or-later", "LGPL-2.0-or-later", "LGPL-2.1-or-later", "LGPL-3.0-or-later")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://api.kde.org/frameworks/kwidgetsaddons/html/index.html"
+    topics = ("conan", "desktop", "widgets", "qt")
+    description = "kwidgetsaddons contains add-on widgets and classes for applications that use the Qt Widgets module."
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False]
+    }
+    default_options = {"shared": False, "fPIC": True}
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def build_requirements(self):
+        self.build_requires("extra-cmake-modules/5.75.0")
+
+    def requirements(self):
+        self.requires("qt/5.15.1")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("kwidgetsaddons-{}".format(self.version), self._source_subfolder)
+
+        tools.replace_in_file(
+            os.path.join(self._source_subfolder, "src", "CMakeLists.txt"),
+            "KDE_INSTALL_DATADIR_KF5",
+            "CONAN_KDE_INSTALL_DATADIR_KF5"
+        )
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.definitions['ECM_MKSPECS_INSTALL_DIR'] = os.path.join(self.package_folder, "res")
+        self._cmake.definitions['KDE_INSTALL_LOGGINGCATEGORIESDIR'] = os.path.join(self.package_folder, "res")
+        self._cmake.definitions['CONAN_KDE_INSTALL_DATADIR_KF5'] = os.path.join(self.package_folder, "res")
+        self._cmake.definitions['CMAKE_INSTALL_LOCALEDIR'] = os.path.join(self.package_folder, "res", "locale")
+        # a hack to avoid installing CMake find modules andconfig files (KB-H016)
+        self._cmake.definitions['KDE_INSTALL_CMAKEPACKAGEDIR'] = os.path.join(self.build_folder, "dummy")
```

Comment:
I have another question about the `KF5` cmake module.
When doing `find_package(KF5 REQUIRED COMPONENTS WidgetsAddons)`,
is the name `KF5` *owned* by kwidgetsaddons, are is this something shared with other kde packages?

Reply:
That is my concern as well, KF5 is the namespace for all kde packages.
In the cmake version one should be able todo

```
find_package(KF5 REQUIRED COMPONENTS WidgetsAddons karchive)
```
or
```
find_package(KF5 REQUIRED COMPONENTS karchive)
find_package(KF5 REQUIRED COMPONENTS WidgetsAddons)
```
So I'm afraid that the approach you mentioned above wont work. I didn't try it thought.

Reply:
I think this deserves an issue on https://github.com/conan-io/conan

Reply:
@SpaceIm @uilianries @danimtb @jgsogo 

Have you encountered something like this before?
Is conan able to handle this?

Reply:
I guess the only way to handle this case is to add a master recipe requiring kde recipes. This master recipe would be the "owner" of KF5 module file.

Reply:
I see this is more or less related to the LLVM one (https://github.com/conan-io/conan-center-index/issues/683) (not sure if these KF5 widgets have some hierarchy among them), but now adding the `names` to the cocktail. With the _names_, at first sight, it looks like the best (only?) approach is the one suggested by @SpaceIm: a master/proxy recipe to choose the modules to _activate_ and provide in the one and only `FindKF5.cmake` file. 🤔

Reply:
This means that the order of executing the `package_info` methods becomes important.
As the *master*'s `package_info` recipe should be executed last (or at least collect after its minors have added themselves to the list).

Reply:
It would be a regular package requiring other packages... the `cmake_find_package` generator will take care of transitively include all the requirements. Not sure what you mean.

Reply:
If what you're proposing is to let a master recipe selectively add requirements using options, that is imho unmaintainable.
For adding a new KDE recipe, you should then make changes to 2 recipes: the recipe itself + the master recipe.
That's way too complicated imho.

Reply:
If qt and boost were modularized, it would require the same feature.

Reply:
I agree with @madebr having a master recipe just so that conan will be able to generate cmake_find_package does not sounds very convenient.
Also, how would the master recipe handle multiple libraries? Someone may require only `kwidgetsaddons`, someone else may require `kwidgetsaddons` and `kpropert` 
If the master recipe will depends on both libraries, this will force users to depend on both even if only one is needed.

Reply:
We can think about a Conan feature that allows to _gather_ different requirements into the same file generated by `cmake_find_package` (or any other generator) if the file happens to use the same. It would require some Conan magic to compose this file... does it works always or it is somehow declared in the consumer recipe? can it work for transitive requirements? As it depends on generator filenames/names then it cannot be resolved in the graph,... I see some questions here that don't have an obvious answer, but maybe we can open an issue in the Conan repo and move this feature-request there.

---

On the other hand, a master/parent recipe is something that can be added right now. I'm not saying it is the best approach, just one alternative. That recipe would need options to define the requirements to use:


```python
class KF5Recipe(ConanFile):
   options = {
      'kwidgetsaddons': [True, False],
      'kpropert': [True, False],
   }

   def requirements(self):
      if self.options.kwidgetsaddons:
         self.requires('kwidgetsaddons/{}'.format(self.version))
      if self.options.kpropert:
         self.requires('kpropert/{}'.format(self.version))

   def package_info(self):
      if self.options.kwidgetsaddons:
         self.cpp_info.components['kwidgetsaddons'].libs = self.deps_cpp_info['kwidgetsaddons'].libs
         self.cpp_info.components['kwidgetsaddons'].includes = self.deps_cpp_info['kwidgetsaddons'].includes
         ...
      if self.options.kpropert:
         self.cpp_info.components['kpropert']...
```

And use it from consumers:

```python

class Consumer(ConanFile):
   requires = "kf5/version"
   default_options = {'kf5:kwidgetsaddons': True, 'kf5:kpropert': False}
```

Only required _components_ will be created and only required _libraries_ will be retrieved and available in the `kf5::` namespace.